### PR TITLE
rh_cloud test coverage

### PR DESCRIPTION
**Pull request by adarshdubey-star**
Test coverage for BZ: https://bugzilla.redhat.com/show_bug.cgi?id=2035204

### CODE REVIEW

**Review by pondrejk — APPROVED** [review]
ack pending test results

**Comment by adarshdubey-star** [comment]
trigger: test-robottelo
pytest: tests/foreman/api/test_rhcloud_inventory.py -k test_positive_tag_values_max_length

File: tests/foreman/api/test_rhcloud_inventory.py [hunk 1]
```diff
@@ -21,6 +21,7 @@
 import pytest
 from fauxfactory import gen_alphanumeric
 from fauxfactory import gen_string
+from nailgun import entities
```
Comment (ogajduse) [inline tests/foreman/api/test_rhcloud_inventory.py:24]: Same as in https://github.com/SatelliteQE/robottelo/pull/9510#discussion_r875714115, please use the Satellite object to access nailgun entities.

File: tests/foreman/api/test_rhcloud_inventory.py [hunk 2]
```diff
@@ -406,3 +407,48 @@ def test_rh_cloud_tag_values(
...
                     break
+
+
+@pytest.mark.run_in_one_thread
+@pytest.mark.tier2
+def test_positive_tag_values_max_length(
+    inventory_settings, organization_ak_setup, rhcloud_registered_hosts, rhcloud_sat_host
+):
+    """Verify that tags values are truncated properly for the host parameter
+            with max length.
```
Comment (ogajduse) [inline tests/foreman/api/test_rhcloud_inventory.py:418]:
```suggestion
    """Verify that tags values are truncated properly for the host parameter
    with max length.
```

File: tests/foreman/api/test_rhcloud_inventory.py [hunk 3]
```diff
@@ -406,3 +407,48 @@ def test_rh_cloud_tag_values(
...
+
+    :Steps:
+        1. Enable include_parameter_tags setting
+        2. Create a host parameter with long text value.
+        3. Generate an rh_cloud reportq
+        4. Observe the tag generated from the parameter.
+
+    :expectedresults:
+        1. Parameter tag value must must not be created after the
+                allowed length.
```
Comment (ogajduse) [inline tests/foreman/api/test_rhcloud_inventory.py:430]: One extra `must` word here. And I suggest to format it differently if there is only single expected result.
```suggestion
    :expectedresults: Parameter tag value must not be created after the allowed length.
```

File: tests/foreman/api/test_rhcloud_inventory.py [hunk 4]
```diff
@@ -406,3 +407,48 @@ def test_rh_cloud_tag_values(
...
+):
+    """Verify that tags values are truncated properly for the host parameter
+            with max length.
+
+    :id: dbcc7245-88af-4c35-87b8-92de01030cb5
+
+    :Steps:
+        1. Enable include_parameter_tags setting
+        2. Create a host parameter with long text value.
+        3. Generate an rh_cloud reportq
```
Comment (ogajduse) [inline tests/foreman/api/test_rhcloud_inventory.py:425]:
```suggestion
        3. Generate a rh_cloud report
```

**Review by ogajduse — CHANGES_REQUESTED** [review]

File: tests/foreman/api/test_rhcloud_inventory.py [hunk 5]
```diff
@@ -406,3 +406,52 @@ def test_rh_cloud_tag_values(
...
+
+
+@pytest.mark.run_in_one_thread
+@pytest.mark.tier2
+def test_positive_tag_values_max_length(
+    inventory_settings,
+    organization_ak_setup,
+    rhcloud_registered_hosts,
+    rhcloud_sat_host,
+    default_sat,
```
Comment (ogajduse) [inline tests/foreman/api/test_rhcloud_inventory.py:418]: Please rebase your branch on the current master branch and change `default_sat` to `target_sat` as there is no `default_sat` fixture anymore.
See @JacobCallahan's email sent out on May 20, 2022, to satqe-list.

Comment (adarshdubey-star) [reply tests/foreman/api/test_rhcloud_inventory.py]: updated!

File: tests/foreman/api/test_rhcloud_inventory.py [hunk 6]
```diff
@@ -406,3 +406,53 @@ def test_rh_cloud_tag_values(
...
+    rhcloud_sat_host,
+    target_sat,
+):
+    """Verify that tags values are truncated properly for the host parameter
+       with max length.
+
+    :id: dbcc7245-88af-4c35-87b8-92de01030cb5
+
+    :Steps:
+        1. Enable include_parameter_tags setting
```
Comment (ColeHiggins2) [inline tests/foreman/api/test_rhcloud_inventory.py:426]: add period for consistency

**Review by ColeHiggins2 — APPROVED** [review]
ACK

**Review by ogajduse — APPROVED** [review]
ACK
Pending comment from @ColeHiggins2. Please do a PRT run if possible.

**Comment by adarshdubey-star** [comment]
> ACK Pending comment from @ColeHiggins2. Please do a PRT run if possible.

I triggered the PRT previously it gets caught with a host teardown error which is not related to this test. The test passes locally.

**Review by jameerpathan111 — APPROVED** [review]

**Pull request merged**